### PR TITLE
Add "Progressive Growing of GANs" (ProGAN) model

### PR DESCRIPTION
**Pull request by seovchinnikov**
I've decided to add a new model to the framework -  an implementation of the paper "Progressive Growing of GANs": https://arxiv.org/abs/1710.10196.
It's basically a GAN that generates images from a random noise with a support of high resolution images by training G and D nets progressively. I've decided to post a PR because found it easily to implement it inside this repo's framework.

>     Model requires dataset of type dataset_mode='single', generator netG='progan', discriminator netD='progan'.
>     Please note that opt.crop_size (default 256) == 4 * 2 ** opt.max_steps (default max_steps is 6).
>     ngf and ndf controlls dimensions of the backbone (128-512).
>     Network G is a master-generator (accumulates weights for eval) and network C (stands for current) is a
>     current trainable generator.
> 
>     See also:
>         https://github.com/tkarras/progressive_growing_of_gans
>         https://github.com/odegeasslbc/Progressive-GAN-pytorch

Please note that PR contains changes from PR https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/pull/1090 with NVIDIA apex.

### CODE REVIEW

**Comment by junyanz** [comment]
Thanks for your contribution. Have you been able to reproduce the FID number on the ProgGAN datasets?

**Comment by seovchinnikov** [comment]
Sorry for a long response, I've only tested it on our private dataset, need help to evaluate it on Celeba

**Comment by aashana94** [comment]
I am trying to run metrics using the original implementation of Progressive Growing of GAN by tkarras but I am not able to, has anyone any luck? or can help me with it? 

https://github.com/tkarras/progressive_growing_of_gans

**Comment by ridhachahed** [comment]
Hey ! You use 2 generators with one being the running average of the other and set adam parameter beta1 to 0. Why this choice ? Why not use the same generator for training and testing ?

**Comment by maxmax1992** [comment]
May I ask why are you in the ProGanModel doing generation from torch.randn and not from source Image A? It seems also that you don't even save the image A as input, this contradicts with the whole cycle consistency of the CycleGan as there's no loss between source image A and reconstruction of A.

**Comment by junyanz** [comment]
We will leave it as is. It might introduce too many changes to add a new GAN to the current repo. But if anyone is interested in progressive gans, you can refer to this PR.
